### PR TITLE
Merge Halfface into VolMesh

### PR DESCRIPTION
**Pull request by tomvanmele**
### What type of change is this?

- [ ] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/api/compas.datastructures.VolMesh.rst [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+    ~VolMesh.halffaces
+    ~VolMesh.halffaces_on_boundaries
+    ~VolMesh.vertex_sample
+    ~VolMesh.vertices
+    ~VolMesh.vertices_on_boundaries
+    ~VolMesh.vertices_where
+    ~VolMesh.vertices_where_predicate
+
+
+Attributes
```
Comment (gonzalocasas) [inline docs/api/compas.datastructures.VolMesh.rst:82]: For this tweaked classes, the `attributes` section can be quite confusing with the normal python attributes. How about we name this `Custom attributes`?

Comment (tomvanmele) [reply docs/api/compas.datastructures.VolMesh.rst:82]: was thinking of splitting up the sections into "Vertex Attributes", "Edge Attributes", ...

Comment (tomvanmele) [reply docs/api/compas.datastructures.VolMesh.rst:82]: a bit like the sections in the code itself

Comment (gonzalocasas) [reply docs/api/compas.datastructures.VolMesh.rst:82]: good idea 👍

**Pull request merged**